### ISSUE TITLE
Review-2018.06.10-刘文杰

### ISSUE THREAD

**Issue by liuwenjie1995** [DIYer22/weekly-review#168]
# 本周完成的工作
1.  日常阅读一篇gan的文章，在网上记录了一个博客，DCGAN，BEGAN，等gan都进行了分析，但是还没写好，存在草稿里，将内容放在草稿箱里，在完成胡老师的任务之后，再进行下一次编辑。

2. 胡老师布置的CAM方法，是一个分类网络，在分类网络的最后一层中，将原来的Dense操作改为了Global average pool，我重新查看了一下几个pool，发现没有，经研究后发现，global average pool是将原来图像的每个点求总平均数得到一个点。这个点带有图像的位置特征，将1024个8 * 8的特征图，1024个1 * 1（由gap pool处理后的图像）依次相乘可以的得到一个结果图，将这个结果图强制放大成为原图大小就可以了。这个结果就是cam处理的结果。我使用了vgg16进行这个网络的配置，并且，导入了vgg16的预训练模型，但是效果都不是很好。

3. 胡老师说cam网络有现成的代码，让我去找找，我看到了几个都是pytorch的代码，不是很好意思说，我不知道为什么装pytorch0.4的版本，无论是pip，conda，源代码安装都有问题，cmake版本总是提示出现错误，但是我的cmake是好的，现在先cam方法就做到这里了。

4. U-Net这里有了vgg16的预训练模型。我重新修改了一下网络结构，将预训练模型导入。在网上看了很多文章，但是没有一个给我讲，在添加了预训练模型之后，图像的输入网络方式是什么，林城龙给我讲的是像素值减去均值，现在的网络是将值除以255，减去1，这一块不是很懂，到底如何放入数据？现在添加了预训练模型之后，训练结果依旧是79%左右，距离现在主流的84~86还是有一定的差距，使用DenseNet？resnet？再试试？现在我认为瓶颈不在网络，而是在于处理图像，learning rate这一块，有机会再考虑一下吧

5.  爬虫爬取了4000张图片，选取了500张最合适的图像。

# 本周学习文章
* 对偶学习和显著性区域分析是老师最近讲ppt的重点，我最近下载了李老师的文章，看了一下显著性那一块乘积的方式，也学习了一下对偶学习。简单来说，对偶学习就是一个两者学习的过程，我不知道我的中文翻译成为英文是否是正确的，所以可以将我的中文翻译成为英文，再从英文翻译过来，若我的句子翻译回中文是正确的，那么我翻译成为英文的方式就是正确的。对偶学习在弱监督这一块意义很大，主要是针对文字处理RNN那一块的，在cnn这一块，我接触到的就是dcgan了。

## 遇到的问题
pytorch 安装问题，现在只能使用pytorch0.3 只能自己写实验，很多网上的代码都是torch0.4
cam效果不好的问题，胡老师说过了，对于一个人脸来说一定要做人脸对齐，背景处理，但是这一块不是我的强项。我会继续看看，问问学长。

**Comment by DIYer22** [comment]
4. Normalize 一操作般是 `x/数据集的std - 数据集的mean` ，在PyTorch 里面使用ImageNet预训练模型时
 Normalize一句就搞定`transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225])` ,这个资料不错[=>PyTorch Transfer Learning tutorial](https://pytorch.org/tutorials/beginner/transfer_learning_tutorial.html#visualize-a-few-images)

* 对偶学习 在cnn的应用应该是 Cycle GAN 之类的吧.......
